crates/wasi-common/src/sync/mod.rs

```diff
@@ -131,7 +131,7 @@ impl WasiCtxBuilder {
 
 pub fn random_ctx() -> Box<dyn RngCore + Send + Sync> {
     let mut rng = cap_rand::thread_rng(cap_rand::ambient_authority());
-    Box::new(cap_rand::rngs::StdRng::from_seed(rng.gen()))
+    Box::new(cap_rand::rngs::StdRng::from_seed(rng.r#gen()))
```

Comment:
cc @sunfishcode on this in case you weren't already aware, but eventually `cap-rand` will probably want to rename (or at least provide an alternative to) its `gen` method to be easier to use in the 2024 edition.